sdk/Lusid.Sdk.Tests/tutorials/Instruments/DemoCreditDefaultSwap.cs

```diff
@@ -0,0 +1,157 @@
+                flowConventions: cdsFlowConventions,
+                couponRate: 0.5m,
+                protectionDetailSpecification: cdsProtectionDetailSpecification,
+                instrumentType: LusidInstrument.InstrumentTypeEnum.CreditDefaultSwap
+            );
+            // ASSERT that it was created
+            Assert.That(cds, Is.Not.Null);
+
+            // CAN NOW UPSERT TO LUSID
+            string uniqueId = Guid.NewGuid().ToString();
+            UpsertOtcInstrumentToLusid(cds,  uniqueId);
+
+            // CAN NOW QUERY FROM LUSID
+            var retrieved = QueryOtcInstrumentFromLusid(uniqueId);
+            Assert.That(retrieved.InstrumentType == LusidInstrument.InstrumentTypeEnum.CreditDefaultSwap);
+            var roundTripCds = retrieved as CreditDefaultSwap;
+            Assert.That(roundTripCds, Is.Not.Null);
+            Assert.That(roundTripCds.CouponRate, Is.EqualTo(cds.CouponRate));
+            Assert.That(roundTripCds.Ticker, Is.EqualTo(cds.Ticker));
+            Assert.That(roundTripCds.MaturityDate, Is.EqualTo(cds.MaturityDate));
+            Assert.That(roundTripCds.StartDate, Is.EqualTo(cds.StartDate));
+            Assert.That(roundTripCds.FlowConventions.Currency, Is.EqualTo(cds.FlowConventions.Currency));
+            Assert.That(roundTripCds.FlowConventions.PaymentFrequency, Is.EqualTo(cds.FlowConventions.PaymentFrequency));
+            Assert.That(roundTripCds.FlowConventions.ResetDays, Is.EqualTo(cds.FlowConventions.ResetDays));
+            Assert.That(roundTripCds.FlowConventions.SettleDays, Is.EqualTo(cds.FlowConventions.SettleDays));
+            Assert.That(roundTripCds.FlowConventions.PaymentCalendars.Count, Is.EqualTo(cds.FlowConventions.PaymentCalendars.Count));
+            Assert.That(roundTripCds.FlowConventions.PaymentCalendars, Is.EquivalentTo(cds.FlowConventions.PaymentCalendars));
+            
+            DeleteItems(null,null,null,uniqueId);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void DemoCreditDefaultSwapValuation(bool inlineValuation)
+        {
+            // CREATE a portfolio with instrument
+            var scope = Guid.NewGuid().ToString();
+            CreditDefaultSwap cds = (CreditDefaultSwap) InstrumentExamples.CreateExampleCreditDefaultSwap();
+
+            UpsertFxRate(scope, EffectiveAt, EffectiveAt, false);
+            UpsertRateCurves(scope, EffectiveAt);
+            UpsertResetQuotes(scope, EffectiveAt.AddDays(-4));
+            
+            // UPSERT CDS spread curve before upserting recipe
+            UpsertCdsSpreadCurves(scope, TestDataUtilities.EffectiveAt, cds.Ticker, cds.FlowConventions.Currency, cds.ProtectionDetailSpecification.Seniority,
+                cds.ProtectionDetailSpecification.RestructuringType); 
+
+            // CALL valuation
+            var valuation = Valuation(cds, scope, ModelSelection.ModelEnum.Discounting, EffectiveAt, inlineValuation);
```

Comment:
I think there is too much obfuscation here on the recipe, a user trying to do a valuation after creating the instrument would be looking for a starting recipe

Reply:
Ok - can look to break it out.